data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/bulk/BulkAPIWrapper.java

```diff
@@ -0,0 +1,92 @@
+package org.opensearch.dataprepper.plugins.sink.opensearch.bulk;
+
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.opensearch._types.ErrorResponse;
+import org.opensearch.client.opensearch._types.OpenSearchException;
+import org.opensearch.client.opensearch.core.BulkRequest;
+import org.opensearch.client.opensearch.core.BulkResponse;
+import org.opensearch.client.transport.JsonEndpoint;
+import org.opensearch.client.transport.endpoints.SimpleEndpoint;
+import org.opensearch.client.util.ApiTypeHelper;
+import org.opensearch.dataprepper.plugins.sink.opensearch.BackendVersion;
+import org.opensearch.dataprepper.plugins.sink.opensearch.index.IndexConfiguration;
+
+import javax.ws.rs.HttpMethod;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class BulkAPIWrapper {
+    static final String DUMMY_DEFAULT_INDEX = "dummy";
+    private final IndexConfiguration indexConfiguration;
+    private final OpenSearchClient openSearchClient;
+
+    public BulkAPIWrapper(final IndexConfiguration indexConfiguration,
+                                final OpenSearchClient openSearchClient) {
+        this.indexConfiguration = indexConfiguration;
+        this.openSearchClient = openSearchClient;
+    }
+
+    public BulkResponse bulk(BulkRequest request) throws IOException, OpenSearchException {
```

Comment:
We should probably split these classes.

Have an interface for `BulkApiWrapper`. Then have a factory which produces either an `Es6BulkApiWrapper` or an `OpenSearchDefaultBulkApiWrapper`.

Then you can drop this conditional.

And you separate the code that is for ES6 entirely from the code for other versions.